### PR TITLE
handle non-utf8 description encodings

### PR DESCRIPTION
**Pull request by mslynch**
Resolves https://github.com/rstudio/rsconnect/issues/974.

### CODE REVIEW

**Review by hadley — APPROVED** [review]
Probably worth a small unit test?

**Comment by mslynch** [comment]
@hadley looks like this breaks on `oldrel-4`. Any suggestions for a path forward?

**Comment by kevinushey** [comment]
Can you just use `packageDescription()`? That tries to handle Encoding internally in a similar way.

**Comment by mslynch** [comment]
`packageDescription()` works beautifully, thanks!

File: R/bundlePackage.R [hunk 1]
```diff
@@ -114,6 +114,6 @@ availablePackages <- function(repos) {
 }
 
 package_record <- function(name, lib_dir = NULL) {
-  path <- system.file("DESCRIPTION", package = name, lib.loc = lib_dir)
-  as.list(as.data.frame(read.dcf(path)))
+  record <- packageDescription(name, lib.loc = lib_dir, encoding = "UTF-8")
+  lapply(record, identity)
```
Comment (hadley) [inline R/bundlePackage.R:118]: Could that just be `as.list(record)`?

Comment (mslynch) [reply R/bundlePackage.R:118]: That gives the error `Error: No method asJSON S3 class: packageDescription`. Not sure if there's a better way around this.

**Pull request merged**